### PR TITLE
Add Dynamic Import Support

### LINKED ISSUE #21: LDRouteGuard - Dynamic Imports
Use Case: LDRouteGuard should support dynamic imports to allow for code splitting

Acceptance Criteria:
- LDRouteGuard should function correctly when imported dynamically
- LDRouteGuard should function correctly when the component passed to it is imported dynamically

### PR DESCRIPTION
**Pull request by benjamin-t-frost**
Story details: https://app.clubhouse.io/dashhudson/story/30993

This is important for maintaining the ability to code split based on route.

### CODE REVIEW

File: src/components/LDRouteGuard.vue [hunk 1]
```diff
@@ -8,14 +8,20 @@ import ldRedirectMixin from '../mixins/ldRedirect';
...
     requiredFeatureFlag: { type: String, required: true },
     to: { type: [String, Object], required: true },
   },
   computed: {
     show() {
       return this.$ld.ready && this.$ld.flags[this.requiredFeatureFlag];
     },
+    isPromise() {
+      return !!this.component && typeof this.component.then === 'function';
+    },
```
Comment (kepheusikarius) [inline src/components/LDRouteGuard.vue]: Why expose this as a property?

Comment (benjamin-t-frost) [reply src/components/LDRouteGuard.vue]: To show the intent of what this check is. Not everyone looking at this is going to realize `.then` implies a promise.

I guess, in that regard, it makes more sense to name it more specifically.

File: src/components/LDRouteGuard.vue [hunk 2]
```diff
@@ -8,14 +8,20 @@ import ldRedirectMixin from '../mixins/ldRedirect';
...
     requiredFeatureFlag: { type: String, required: true },
     to: { type: [String, Object], required: true },
   },
   computed: {
     show() {
       return this.$ld.ready && this.$ld.flags[this.requiredFeatureFlag];
     },
+    componentIsPromise() {
+      return !!this.component && typeof this.component.then === 'function';
+    },
```
Comment (kepheusikarius) [inline src/components/LDRouteGuard.vue]: Sorry, I wasn't clear. Let me try again...

Why expose this as a property instead of using a local variable in the one place it's used?

Comment (benjamin-t-frost) [reply src/components/LDRouteGuard.vue]: No reason in particular. I'll change ~

**Review by nathanmyles — APPROVED** [review]

**Review by kepheusikarius — APPROVED** [review]
